addons/website_event_booth_sale/models/sale_order.py

```diff
@@ -42,9 +42,36 @@ def _website_product_id_change(self, order_id, product_id, qty=0, **kwargs):
                     event_booth_registrations_command = new_registrations_commands
                 values['event_booth_registration_ids'] = event_booth_registrations_command
 
+            discount = 0
+            order = self.env['sale.order'].sudo().browse(order_id)
+            booth_currency = booths[:1].product_id.currency_id
+            pricelist_currency = order.pricelist_id.currency_id
+            price = sum(booth.booth_category_id.price for booth in booths)
+            price_reduce = sum(booth.booth_category_id.price_reduce for booth in booths)
+            if booth_currency != pricelist_currency:
+                price = booth_currency._convert(
+                    price,
+                    pricelist_currency,
+                    order.company_id,
+                    order.date_order or fields.Datetime.now()
+                )
+                price_reduce = booth_currency._convert(
+                    price_reduce,
+                    pricelist_currency,
+                    order.company_id,
+                    order.date_order or fields.Datetime.now()
+                )
+            if order.pricelist_id.discount_policy == 'without_discount':
+                price_unit = price
+                discount = (price - price_reduce) / price * 100
+                if discount < 0:
+                    discount = 0
+            else:
+                price_unit = price_reduce
             values.update(
                 event_id=booths.event_id.id,
-                price_unit=sum(booth.booth_category_id.price_reduce for booth in booths),
+                discount=discount,
+                price_unit=price_unit,
```

Comment:
While you are modifying the logic, I suppose taxes and fpos should also be considered here.

(maybe to confirm with PO, but considering the recent bugfixes in the sale scope, I feel like this would be the next bugfix step :P)